CMU-313/spring23-nodebb-green
# [User Story 2] Update backend functionality for resolved field & added a "resolved" button

## Description
Resolves #5 #4 
* Fixed the `PUT` endpoint for resolving fields from @oeb01's work
* Builds on @aadenuga's work to add a call to the new `PUT` endpoint upon clicking on a button
* Updates permissions so that topic owners can resolve posts that belong to them
* Refactored `src/topics/data.js` to parse boolean values in more extensible way

https://user-images.githubusercontent.com/50491000/218745772-7c7cebc7-7c6a-4694-93b0-68424f45505b.mp4


## Motivation and Context
Patches some of the bugs & finishes up the functionality for user story 2 ("As an instructor, I want to mark questions as resolved, so that I can tell which questions are still unanswered.")

## How Has This Been Tested?
* Created private post to check that private posting still works after merging in main
* Testing `resolve` works:
    1. Check in api that `resolve` is marked as false initially
    2. Press `resolve`
    3. Check in api that `resolve` is now marked as true
* Checking `resolve` permissions:
    * Owners, admins, & mods see the `resolve` button
    * Others don't see button
    * 403 error when non-admins/mods try to make a resolve fetch request

## Types of changes




## Checklist:



-   [ ] My change requires a change to the README documentation

## Next steps
- Add some indication on the interface that post has been resolved (Issue #6)
- Create banner that gives user feedback that post was resolved
    - Edit the `handleResolve`'s event listener callback function in public/src/client/topic.js to do so
- TypeScript translations
